recipes/ncbi-cxx-toolkit-public/26.0/conanfile.py

```diff
@@ -0,0 +1,1450 @@
+    def requirements(self):
+        NCBIreqs = self._get_RequiresMapKeys()
+        for req in NCBIreqs:
+            pkg = self._translate_ReqKey(req)
+            if pkg is not None:
+                self.requires(pkg)
+
+#----------------------------------------------------------------------------
+    def source(self):
+        tk_url = self.conan_data["sources"][self.version]["url"]
+        print("getting Toolkit sources...")
+        print("from url: " + tk_url)
+        tools.get(**self.conan_data["sources"][self.version], strip_root = True)
+
+#----------------------------------------------------------------------------
+    def imports(self):
+        self.copy("license*", dst="licenses", folder=True, ignore_case=True)
+
+#----------------------------------------------------------------------------
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.configure(source_folder=self._source_subfolder)
+# Visual Studio sometimes runs "out of heap space"
+        if self.settings.compiler == "Visual Studio":
+            self.run("cmake --build . %s -j 1" % cmake.build_config)
+        else:
+            cmake.build()
+
+#----------------------------------------------------------------------------
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+
+#----------------------------------------------------------------------------
+    def package_info(self):
+        if self.settings.os == "Windows":
+            self.cpp_info.components["ORIGLIBS"].system_libs = ["ws2_32", "dbghelp"]
+            self.cpp_info.components["NETWORKLIBS"].system_libs = [""]
+        elif self.settings.os == "Linux":
+            self.cpp_info.components["ORIGLIBS"].system_libs = ["dl", "rt", "m", "pthread"]
+            self.cpp_info.components["NETWORKLIBS"].system_libs = ["resolv"]
+        elif self.settings.os == "Macos":
+            self.cpp_info.components["ORIGLIBS"].system_libs = ["dl", "c", "m", "pthread"]
+            self.cpp_info.components["NETWORKLIBS"].system_libs = ["resolv"]
+
+        NCBIreqs = self._get_RequiresMapKeys()
+        for req in NCBIreqs:
+            pkg = self._translate_ReqKey(req)
+            if pkg is not None:
+                pkg = pkg[:pkg.find("/")]
+                ref = pkg + "::" + pkg
+                self.cpp_info.components[req].requires = [ref]
+                self.cpp_info.components["ORIGLIBS"].requires.append(ref)
+            else:
+                self.cpp_info.components[req].libs = []
+
+        allexports = {}
+        impfile = self.package_folder + "/res/ncbi-cpp-toolkit.imports"
+        if os.path.isfile(impfile):
+            allexports = set(open(impfile).read().split())
+
+#============================================================================
+        if self.settings.os == "Windows" and self.options.shared:
+#12--------------------------------------------------------------------------
```

Comment:
Lot of components, but only for windows shared?

It could be one line per component with some factorization.

Reply:
yes, only for window/shared. this is by design of the toolkit